pkg/sql/logictest/testdata/logic_test/join

```diff
@@ -1112,3 +1112,11 @@ statement ok
 DROP TABLE empty;
 DROP TABLE fk_ref;
 DROP TABLE xy;
+
+statement ok
+CREATE TABLE t34524 (a INT PRIMARY KEY)
+
+# Regression test for #34524.
+query I
+(SELECT NULL FROM t34524) EXCEPT (VALUES((SELECT 1 FROM t34524 LIMIT 1)), (1))
+----
```

Comment:
nit: I think the `pkg/sql/logictest/testdata/logic_test/union` might be a better place for this since there are no explicit joins in the query. Really, `pkg/sql/logictest/testdata/logic_test/set` seems best, but most `EXCEPT` tests are currently in the `union` file...